scripts/tests/twister_blackbox/test_hardwaremap.py

```diff
@@ -0,0 +1,318 @@
+        )
+    ]
+    TESTDATA_3 = [
+        (
+            'Texas Instruments',
+            'DAPLink CMSIS-DAP',
+            'abcd','las'
+        ),
+        (
+            'Texas Instruments',
+            'DAPLink CMSIS-DAP',
+            'abcd','dse0'
+        )
+    ]
+
+
+    @classmethod
+    def setup_class(cls):
+        apath = os.path.join(ZEPHYR_BASE, 'scripts', 'twister')
+        cls.loader = importlib.machinery.SourceFileLoader('__main__', apath)
+        cls.spec = importlib.util.spec_from_loader(cls.loader.name, cls.loader)
+        cls.twister_module = importlib.util.module_from_spec(cls.spec)
+
+
+    @classmethod
+    def teardown_class(cls):
+        pass
+
+
+    @pytest.mark.usefixtures("clear_log")
+    @pytest.mark.parametrize(
+        ('manufacturer','product','serial','runner'),
+        TESTDATA_1,
+    )
+
+    def test_generate(self, capfd, manufacturer, product, serial, runner):
+        file_name = "test-map.yaml"
+        path = os.path.join(ZEPHYR_BASE, file_name)
+        args = ['--generate-hardware-map', file_name]
+
+
+        if os.path.exists(path):
+            os.remove(path)
+
+        def mocked_comports():
+            return [
+                mock.Mock(device='/dev/ttyUSB23', \
+                    manufacturer= id_man, \
+                    product = id_pro, \
+                    serial_number = id_serial
+                    )
+                    ]
+
+        for id_man in manufacturer:
+            for id_pro in product:
+                for id_serial in serial:
+                    with mock.patch.object(sys, 'argv', [sys.argv[0]] + args), \
+                        mock.patch('serial.tools.list_ports.comports', \
+                        side_effect=mocked_comports), \
+                        pytest.raises(SystemExit) as sys_exit:
+                        self.loader.exec_module(self.twister_module)
+
+                    out, err = capfd.readouterr()
+                    sys.stdout.write(out)
+                    sys.stderr.write(err)
+
+                    assert os.path.exists(path)
+
+                    expected_data = '- connected: true\n'+\
+                                    f'  id: {id_serial}\n'+\
+                                    '  platform: unknown\n'+\
+                                    f'  product: {id_pro}\n'+\
+                                    f'  runner: {runner}\n'+\
+                                    '  serial: /dev/ttyUSB23\n'
+
+                    load_data = open('test-map.yaml').read()
+                    assert load_data == expected_data
+
+                    if os.path.exists(path):
+                        os.remove(path)
+
+                    assert str(sys_exit.value) == '0'
+                    loggers = [logging.getLogger()] \
+                        + list(logging.Logger.manager.loggerDict.values()) \
+                        + [logging.getLogger(name) for \
+                                    name in logging.root.manager.loggerDict]
```

Comment:
Previously, we've had [plus] [newline], now it's [newline] [plus] - probably should be normalised.

Reply:
fixed!